sap/rfc/user.py

```diff
@@ -410,14 +413,15 @@ def create_user(self, connection, user_builder: UserBuilder) -> BAPIReturn:
         """Creates a new user for the given user data"""
 
         rfc_ret = self._call_bapi_method(connection, 'BAPI_USER_CREATE1', user_builder.build_rfc_params())
-        result = rfc_ret['RETURN']
+        bapi_result = BAPIReturn(rfc_ret['RETURN'])
+        if bapi_result.is_error:
+            return bapi_result
 
-        if not BAPIReturn(result).is_error:
-            rfc_password_change_ret = self._change_user_password(connection, user_builder)
-            if rfc_password_change_ret:
-                result.append(rfc_password_change_ret)
+        rfc_password_change_ret = self._change_user_password(connection, user_builder)
+        if rfc_password_change_ret:
+            rfc_ret['RETURN'].append(rfc_password_change_ret)
 
-        return BAPIReturn(result)
+        return BAPIReturn(rfc_ret['RETURN'])
```

Comment:
Just return the already created instance `bapi_result`. Or am I missing something?

Reply:
I cannot return just `bapi_result` because I am also appending there messages from the second BAPI call.

Reply:
Ah, I missed that. I am not sure it's the clearest possible solution. What about adding a helper variable holding the list of all BAPI result lines? Accessing `rfc_ret['RETURN']` is neither efficient (as we must search the dict) nor safe because the list can be immutable/

Reply:
Agree, basically making it like it was in the first commit ;-) 

A new helper variable called `bapi_messages` was added.